TCAT/Controllers/RouteOptionsViewController.swift

```diff
@@ -152,23 +151,19 @@ class RouteOptionsViewController: UIViewController, UITableViewDelegate, UITable
 
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
-        
-        takedownReachability()
-        if isBannerShown {
-            banner?.dismiss()
-            banner = nil
-            isBannerShown = false
-        }
-        
+        // Takedown reachability
+        reachability?.stopNotifier()
+        NotificationCenter.default.removeObserver(self, name: .reachabilityChanged, object: reachability)
+        // Remove banner
+        banner?.dismiss()
+        banner = nil
         // Deactivate and remove timers
-        for (_, timer) in timers {
-            timer.invalidate()
-        }
+        for (_, timer) in timers { timer.invalidate() }
```

Comment:
timers.values.forEach { $0.invalidate() }

Reply:
Ah I tried to do something like that, values was what I'm looking for, thanks!

Reply:
Resolved